tests/test_memory_size.cpp

```diff
@@ -0,0 +1,309 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <multipass/exceptions/invalid_memory_size_exception.h>
+#include <multipass/memory_size.h>
+
+#include <gtest/gtest.h>
+
+#include <string>
+
+namespace mp = multipass;
+
+using namespace testing;
+
+namespace
+{
+constexpr auto kilo = 1024LL; // LL: giga times value higher than 4 would overflow if we used only 4bytes here
+constexpr auto mega = kilo * kilo;
+constexpr auto giga = kilo * mega;
+} // namespace
+
+TEST(MemorySize, defaultConstructsToZero)
+{
+    EXPECT_EQ(mp::MemorySize{}.in_bytes(), 0LL);
+}
+
+TEST(MemorySize, interpretsKB)
+{
+    constexpr auto val = 1024;
+    const auto size = mp::MemorySize{std::to_string(val) + "KB"};
+
+    EXPECT_EQ(size.in_bytes(), val * kilo);
+}
+
+TEST(MemorySize, interpretsK)
+{
+    constexpr auto val = 1024;
+    const auto size = mp::MemorySize{std::to_string(val) + "K"};
+
+    EXPECT_EQ(size.in_bytes(), val * kilo);
+}
+
+TEST(MemorySize, interpretsMB)
+{
+    constexpr auto val = 1024;
+    const auto size = mp::MemorySize{std::to_string(val) + "MB"};
+
+    EXPECT_EQ(size.in_bytes(), val * mega);
+}
+
+TEST(MemorySize, interpretsM)
+{
+    constexpr auto val = 1;
+    const auto size = mp::MemorySize{std::to_string(val) + "M"};
+
+    EXPECT_EQ(size.in_bytes(), val * mega);
+}
+
+TEST(MemorySize, interpretsGB)
+{
+    constexpr auto val = 1024;
+    const auto size = mp::MemorySize{std::to_string(val) + "GB"};
+
+    EXPECT_EQ(size.in_bytes(), val * giga);
+}
+
+TEST(MemorySize, interpretsG)
+{
+    constexpr auto val = 5;
+    const auto size = mp::MemorySize{std::to_string(val) + "G"};
+
+    EXPECT_EQ(size.in_bytes(), val * giga);
+}
+
+TEST(MemorySize, interpretsNoUnit)
+{
+    constexpr auto val = 1024;
+    const auto size = mp::MemorySize{std::to_string(val)};
+
+    EXPECT_EQ(size.in_bytes(), val);
+}
+
+TEST(MemorySize, interpretsB)
+{
+    constexpr auto val = 123;
+    const auto size = mp::MemorySize{std::to_string(val) + "B"};
+
+    EXPECT_EQ(size.in_bytes(), val);
+}
+
+TEST(MemorySize, interprets0)
+{
+    constexpr auto val = 0;
+    const auto size = mp::MemorySize{std::to_string(val)};
+
+    EXPECT_EQ(size.in_bytes(), val);
+}
+
+TEST(MemorySize, interprets0B)
+{
+    constexpr auto val = 0;
+    const auto size = mp::MemorySize{std::to_string(val) + "B"};
+
+    EXPECT_EQ(size.in_bytes(), val);
+}
+
+TEST(MemorySize, interprets0K)
+{
+    constexpr auto val = 0;
+    const auto size = mp::MemorySize{std::to_string(val) + "K"};
+
+    EXPECT_EQ(size.in_bytes(), val);
+}
+
+TEST(MemorySize, interprets0M)
+{
+    constexpr auto val = 0;
+    const auto size = mp::MemorySize{std::to_string(val) + "M"};
+
+    EXPECT_EQ(size.in_bytes(), val);
+}
+
+TEST(MemorySize, interprets0G)
+{
+    constexpr auto val = 0;
+    const auto size = mp::MemorySize{std::to_string(val) + "G"};
+
+    EXPECT_EQ(size.in_bytes(), val);
+}
```

Comment:
No luck with parameterized tests?

Reply:
Heh, I had, but in other places :P 

Yeah... I agree this would be the exact sort of place to have them. In my defense, I just wanted to be quick about it and so transferred existing tests from the previous "mem-sanitizing" approach with minimal change (multiple cursors in sublime make that easier). I think I wrote those tests before the ones I referred to when I talked about parameterized tests. But let me do that now.